waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -188,43 +189,72 @@ method init*(wf: WakuFilter) =
   wf.handler = handle
   wf.codec = WakuFilterCodec
 
-proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler): T =
+proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler,timeout: Duration = WakuFilterTimeout): T =
   let rng = crypto.newRng()
   var wf = WakuFilter(rng: rng,
                       peerManager: peerManager, 
-                      pushHandler: handler)
+                      pushHandler: handler,
+                      timeout: timeout)
   wf.init()
-  
   return wf
 
 proc setPeer*(wf: WakuFilter, peer: RemotePeerInfo) =
   wf.peerManager.addPeer(peer, WakuFilterCodec)
   waku_filter_peers.inc()
 
+#clear the failed peer table if subscriber was able to connect.
+proc handleClientSuccess*(wf: WakuFilter, subscribers: seq[Subscriber]){.raises: [Defect, KeyError], async.} = 

```

Comment:
Any reason these are public (`*`) and `async`? Afaics this can be done synchronously? In general, functions and procs should be private, unless there's a specific reason to expose this on the public API.

Reply:
resolves in d91eaeb6f12c7deb56d65fb0e275f119d2e56158